src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2218,6 +2218,17 @@ public static bool IsIsOrAsContext(this SyntaxTree syntaxTree, int position, Syn
             // is/as are valid after expressions.
             if (token.IsLastTokenOfNode<ExpressionSyntax>())
             {
+                // Not after lambdas, anonymous methods, or
+                // expressions of type 'void' or 'method group'
+                var expression = token.GetAncestors<ExpressionSyntax>().LastOrDefault();
+                var leftHandBinding = semanticModel.GetSymbolInfo(expression);
+                var symbol = leftHandBinding.GetBestOrAllSymbols().FirstOrDefault();
```

Comment:
Should we do this only if *all* symbols are void?  If there are any in the candidate set that are non-void, you might want the as/is there.